test/support/application_fixture_helper.rb

```diff
@@ -0,0 +1,75 @@
+# typed: ignore
+# frozen_string_literal: true
+
+module ApplicationFixtureHelper
+  TEMP_FIXTURE_DIR = ROOT.join("tmp", "fixtures").to_s
+  DEFAULT_TEMPLATE = :minimal
+
+  def setup_application_fixture
+    @old_working_dir = Dir.pwd
+  end
+
+  def teardown_application_fixture
+    Dir.chdir(@old_working_dir)
+    FileUtils.remove_entry(@app_dir, true) if defined? @app_dir
+  end
+
+  def copy_template(template)
+    copy_dir("test/fixtures/#{template}")
+    Dir.chdir(app_dir)
+  end
+
+  def app_dir
+    copy_template(DEFAULT_TEMPLATE) unless defined? @app_dir
+    @app_dir
+  end
+
+  def config
+    @config ||= Packwerk::Configuration.from_path(app_dir)
+  end
+
+  def path_to(relative_path)
+    File.join(app_dir, relative_path)
+  end
+
+  def paths_to(*relative_paths)
+    relative_paths.map { |path| path_to(path) }
+  end
+
+  def merge_into_yaml_file(relative_path, hash)
+    path = path_to(relative_path)
+    FileUtils.mkpath(File.dirname(path))
+    FileUtils.touch(path)
+    data = YAML.load_file(path) || {}
+    recursive_merge!(data, hash)
+    File.open(path, "w") { |f| YAML.dump(data, f) }
```

Comment:
\nit I think it would be nice to abstract the method into multiple smaller private methods, which makes it easier to read. Something like:

```ruby
def merge_into_yaml_file(relative_path, hash)
    data = read_yaml(path)
    merged_data = recursive_merge(data, hash)
    write_yaml(path)
end

def read_yaml(relative_path)
    path = path_to(relative_path)
    FileUtils.mkpath(File.dirname(path))
    FileUtils.touch(path)
    YAML.load_file(path) || {}
end

def write_yaml(path)
    File.open(path, "w") { |f| YAML.dump(data, f) }
end
```

Thinking of it a bit more, we could potentially move that functionality into a separate YamlFile class since that is the data we are operating on. I think it would make the module a bit easier to understand.

```ruby
class YamlFile
  attr_reader :path

  def initialize(path)
    @path = path
  end

  def merge!(hash)
    merged_data = recursive_merge(read, hash)
    write(merged_data)
  end

  private
  
  def read
     path = path_to(relative_path)
     FileUtils.mkpath(File.dirname(path))
     FileUtils.touch(path)
     YAML.load_file(path) || {}
  end

  def write(data)
    File.open(path, "w") { |f| YAML.dump(data, f) }
  end
```

Reply:
I like your approach with a separate class.  I'll try that out.